src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -176,7 +176,7 @@ private function registerHandlers(ContainerBuilder $container, array $busIds)
             $container->register($locatorId = $bus.'.messenger.handlers_locator', HandlersLocator::class)
                 ->setArgument(0, $handlersLocatorMappingByBus[$bus] ?? array())
             ;
-            if ($container->has($handleMessageId = $bus.'.middleware.handle_message')) {
+            if ($container->has($handleMessageId = $bus.'.middleware.handle_message') || $container->has($handleMessageId = 'messenger.middleware.handle_message')) {
```

Comment:
IIUC this fixes the case where you register a custom middleware on a bus and that middleware service extends the built-in `messenger.middleware.handle_message` abstract service, then it doesn't get the HandlersLocator injected, right?
Here we inject a bus-specific HandlersLocator to the abstract `messenger.middleware.handle_message` service itself, which seems fragile to me as it makes it no more truly abstract: it is then tight to the bus on which the custom middleware is configured. Thus registering e.g. another similar custom `handle_message` middleware on another bus would break the first one because the argument would change at the abstract definition level.
If I correctly understand the intend, I think it's just not supported actually, and I don't think we want to solve it this way.